File: 0014-longest-common-prefix/0014-longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        ans = strs[0]
        for i in strs[1::]:
            while not i.startswith(ans):
                ans = ans[:-1]
                if ans == "":
                    break
        return ans

File: LeetCode/Easy/0014-longest-common-prefix/0014-longest-common-prefix.java
import java.util.*;
class Solution {
    public String longestCommonPrefix(String[] strs) {
        for(int i = 0; i < strs[0].length(); i++)
        {
            for(int j = 0; j < strs.length; j++)
            {
                if(strs[j].length() <= i || strs[0].charAt(i) != strs[j].charAt(i))
                {
                    return strs[0].substring(0,i); 
                }
            }
        }
        return "";
    }
}